frontend/src/pages/PipelineDetails.tsx

```diff
@@ -199,51 +231,68 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     let selectedVersion: ApiPipelineVersion | undefined;
     let versions: ApiPipelineVersion[] = [];
 
-    // If fromRunId is specified, load the run and get the pipeline template from it
-    if (fromRunId) {
+    // If fromRunId or fromRecurringRunId is specified, load the run and get the pipeline template from it
+    if (origin) {
+      const errorTextAdjective = origin.isRecurring ? 'recurring ' : '';
       try {
-        const runDetails = await Apis.runServiceApi.getRun(fromRunId);
+        origin.recurringRun = origin.isRecurring
+          ? await Apis.jobServiceApi.getJob(origin.recurringRunId!)
+          : undefined;
```

Comment:
Should it be `null` instead? My impression is `null` is intentional empty, which is preferred in explicit initialization.

Reply:
Probably the inline expression is a little confusing. We don't have to make the field empty intentionally. I will change this part as
```
if (origin.isRecurring) {
    origin.recurringRun = await Apis.jobServiceApi.getJob(origin.recurringRunId!);
} else {
    origin.run = await Apis.runServiceApi.getRun(origin.runId!);
}
```
(i.e. only define the field when it's needed)